src/Popover/Popover.tsx

```diff
@@ -0,0 +1,145 @@
+  /**
+   * Control whether the Popover should be displayed on the page.
+   */
+  isVisible: boolean;
+  /**
+   * The optional content that will appear as the heading of the Popover
+   */
+  title?: ReactNode;
+  /**
+   * The Optional  offset that should be applied to the left side of the the
+   * Popover element, relative to its placement in the DOM.
+   * Should specified in pi, em, rem, % or 0.
+   */
+  xOffset?: string;
+  /**
+   * The Optional  offset that should be applied to the bottom of the the
+   * Popover element, relative to its placement in the DOM.
+   * Should specified in pi, em, rem, % or 0.
+   */
+  yOffset?: string;
+}
+
+/**
+ * Sets the amount of offset that should be applied to the Popover element
+ */
+interface PopoverWrapperProps {
+  /**
+   * Sets the position of the left edge of the Popover
+   */
+  left: PopoverProps['xOffset'];
+  /**
+   * Sets the position of the bottom edge of the Popover
+   */
+  bottom: PopoverProps['yOffset'];
+}
+
+const PopoverPlaceholder = styled.div`
+  position: absolute;
+  width: 0px;
+  height: 0px;
+`;
+
+/**
+ * Applies styles to the PopoverWrapper, using cascading styles to style the
+ * internal title and body. This allows for restyling the entire componed by
+ * wrapping it, e.g. styled(Popover)
+ */
+const PopoverWrapper = styled.div<PopoverWrapperProps>`
+  border: ${fromTheme('border', 'hairline')};
+  box-shadow: ${fromTheme('shadow', 'xlight')};
+  position: absolute;
+  left: ${({ left }) => left};
+  bottom: ${({ bottom }) => bottom};
+  color: ${fromTheme('color', 'text', 'dark')};
+  max-width: 10em;
+  z-index: 10;
+  > strong {
+    background-color: ${fromTheme('color', 'background', 'subtle')};
+    display: block;
+    font-family: ${fromTheme('font', 'bold', 'family')};
+    font-size: ${fromTheme('font', 'bold', 'size')};
+    font-weight: ${fromTheme('font', 'bold', 'weight')};
+    border-bottom: ${fromTheme('border', 'hairline')};
+    padding: ${fromTheme('ws', 'xsmall')};
+  }
+  > p {
+    background-color: ${fromTheme('color', 'background', 'light')};
+    font-family: ${fromTheme('font', 'note', 'family')};
+    font-weight: ${fromTheme('font', 'note', 'weight')};
+    font-size: ${fromTheme('font', 'note', 'size')};
+    padding: ${fromTheme('ws', 'xsmall')};
+    text-align: left;
+  }
+`;
+
+/**
+ * Renders a small popup notification next to an element on the page.
+ * By default the Popover will appear directly above and to the right of its
+ * location in the DOM, but the xOffset and yOffset props can be used to move
+ * it elsewhere
+ */
+
+const Popover: FunctionComponent<PopoverProps> = ({
+  children,
+  className,
+  forwardRef,
+  isVisible,
+  title,
+  xOffset,
+  yOffset,
+}) => (
+  <PopoverPlaceholder>
+    {isVisible
+      ? (
+        <PopoverWrapper
+          className={className}
+          left={xOffset}
+          bottom={yOffset}
+        >
+          {title ? <strong>{title}</strong> : null}
```

Comment:
I wonder if `strong` is the most semantically appropriate element for a title here? Would a `h5` or` h6` or something be better? [Bootstrap's example](https://getbootstrap.com/docs/4.0/components/popovers/) uses an `h3` for this....(though admittedly, that's likely inserted by the component consumer, rather than the component itself)

Reply:
I did consider using an `h#` component here, but my concern is that there's no way to know where in the page hierarchy this would be inserted, and I figured it's worse semantically to use headings out of order, and because these are generally going to be temporary elements, it could cause problems for a generated TOC to have these as headings that appear and disappear.

Reply:
That makes a ton of sense.